MSVCRT build warning when building Windows App SDK (WinUI) release binaries with native AOT

### Package

Sentry

### .NET Flavor

.NET

### .NET Version

8.0.12

### OS

Windows

### SDK Version

5.0.1

### Self-Hosted Sentry Version

_No response_

### Steps to Reproduce
(Please find a minimal project attached below.)

1. In Visual Studio 2022, create a `Blank App, Packaged (WinUI 3 in Desktop)`, and add Sentry 5.0.1 as a NuGet package. Set the project to Release mode and apply Native AoT (right-click project > Properties > Build > Publish > enable Publish native AOT).
2. Add the [WinUI (with trimming) code](https://docs.sentry.io/platforms/dotnet/guides/winui/#configure-with-trimming) to the project.
3. Package the project (right-click project > Package and publish > Create App Packages, and create a Release x64 MSIX binary).

During the Release build, after mentioning `Generating native code`, the build log will output the following warning: `LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library`

I was able to discover this by using PowerShell:

```
cd "C:\Users\User\source\repos\Project\src\Project"

Get-ChildItem -Recurse | Select-String "msvcrt" -List | Select Path | Format-List *

Path : C:\Users\User\source\repos\Project\src\Project\bin\ARM64\Release\net8.0-windows10.0.26100.0\win-arm64\Sentry.dll

Path : C:\Users\User\source\repos\Project\src\Project\bin\Release\net8.0-windows10.0.26100.0\win-arm64\publish\Sentry.dll

Path : C:\Users\User\source\repos\Project\src\Project\bin\Release\net8.0-windows10.0.26100.0\win-x64\publish\project.exe

Path : C:\Users\User\source\repos\Project\src\Project\bin\Release\net8.0-windows10.0.26100.0\win-x64\publish\project.pdb

Path : C:\Users\User\source\repos\Project\src\Project\bin\x64\Debug\net8.0-windows10.0.26100.0\win-x64\Sentry.dll

Path : C:\Users\User\source\repos\Project\src\Project\bin\x64\Debug\net8.0-windows10.0.26100.0\win-x64\AppX\Sentry.dll

Path : C:\Users\User\source\repos\Project\src\Project\bin\x64\Release\net8.0-windows10.0.26100.0\win-x64\Sentry.dll

Path : C:\Users\User\source\repos\Project\src\Project\bin\x64\Release\net8.0-windows10.0.26100.0\win-x64\AppX\Sentry.dll

Path : C:\Users\User\source\repos\Project\src\Project\bin\x64\Release\net8.0-windows10.0.26100.0\win-x64\native\project.exe

Path : C:\Users\User\source\repos\Project\src\Project\bin\x64\Release\net8.0-windows10.0.26100.0\win-x64\native\project.pdb

Path : C:\Users\User\source\repos\Project\src\Project\obj\ARM64\Release\net8.0-windows10.0.26100.0\win-arm64\linked\Sentry.dll

Path : C:\Users\User\source\repos\Project\src\Project\obj\ARM64\Release\net8.0-windows10.0.26100.0\win-arm64\R2R\Sentry.dll

Path : C:\Users\User\source\repos\Project\src\Project\obj\x64\Release\net8.0-windows10.0.26100.0\win-x64\linked\Sentry.dll

Path : C:\Users\User\source\repos\Project\src\Project\obj\x64\Release\net8.0-windows10.0.26100.0\win-x64\native\project.obj

Path : C:\Users\User\source\repos\Project\src\Project\obj\x64\Release\net8.0-windows10.0.26100.0\win-x64\R2R\Sentry.dll
```



### Expected Result

No warning should appear in the build log.

### Actual Result

`LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library`

[Minimal project (SentryAotTest.zip)](https://github.com/user-attachments/files/18627063/SentryAotTest.zip)

I've just hit this myself, but with native AoT for an ASP.NET Core application. It also fails to compile on Linux, but with a different error. macOS builds with no errors or warnings.

Here's the PR adding Sentry, which will show the diff and link off to the failing GitHub Actions builds: https://github.com/martincostello/website/pull/2504

## Windows

```text
  Website -> D:\a\website\website\artifacts\bin\Website\release\Website.dll
  Generating native code
LINK : error LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library [D:\a\website\website\src\Website\Website.csproj]
D:\.nuget\packages\microsoft.dotnet.ilcompiler\9.0.4\build\Microsoft.NETCore.Native.targets(380,5): error MSB3073: The command ""C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\bin\Hostx64\x64\link.exe" @"D:\a\website\website\artifacts\obj\Website\release\native\link.rsp"" exited with code -1. [D:\a\website\website\src\Website\Website.csproj]
```

## Linux

```text
Website -> /home/runner/work/website/website/artifacts/bin/Website/release/Website.dll
  Generating native code
  /usr/bin/ld.bfd: cannot find -lcurl: No such file or directory
clang : error : linker command failed with exit code 1 (use -v to see invocation) [/home/runner/work/website/website/src/Website/Website.csproj]
/home/runner/.nuget/packages/microsoft.dotnet.ilcompiler/9.0.4/build/Microsoft.NETCore.Native.targets(376,5): error MSB3073: The command ""clang" "/home/runner/work/website/website/artifacts/obj/Website/release/native/Website.o" -o "/home/runner/work/website/website/artifacts/bin/Website/release/native/Website" -Wl,--version-script=/home/runner/work/website/website/artifacts/obj/Website/release/native/Website.exports -Wl,--export-dynamic -gz=zlib -fuse-ld=bfd /home/runner/.nuget/packages/sentry/5.5.1/buildTransitive/../sentry-native/linux-x[64](https://github.com/martincostello/website/actions/runs/14430046290/job/40463832716?pr=2504#step:10:65)/libsentry-native.a /home/runner/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.4/sdk/libbootstrapper.o /home/runner/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.4/sdk/libRuntime.ServerGC.a /home/runner/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.4/sdk/libeventpipe-enabled.a /home/runner/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.4/sdk/libRuntime.VxsortEnabled.a /home/runner/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.4/sdk/libstandalonegc-disabled.a /home/runner/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.4/sdk/libstdc++compat.a /home/runner/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.4/framework/libSystem.Native.a /home/runner/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.4/framework/libSystem.IO.Compression.Native.a /home/runner/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.4/framework/libSystem.Net.Security.Native.a /home/runner/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.4/framework/libSystem.Security.Cryptography.Native.OpenSsl.a /home/runner/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.4/sdk/libz.a -g -Wl,-rpath,'$ORIGIN' -Wl,--build-id=sha1 -Wl,--as-needed -pthread -lcurl -ldl -lrt -lm -pie -Wl,-pie -Wl,-z,relro -Wl,-z,now -Wl,--eh-frame-hdr -Wl,--discard-all -Wl,--gc-sections" exited with code 1. [/home/runner/work/website/website/src/Website/Website.csproj]
```

### macOS

```text
  Website -> /Users/runner/work/website/website/artifacts/bin/Website/release/Website.dll
  Generating native code
  Website -> /Users/runner/work/website/website/artifacts/publish/Website/release/
```

After bumping to Native SDK v0.8.3 we're actually seeing the same problem in our [integration tests](https://github.com/getsentry/sentry-dotnet/actions/runs/14229496567/job/40146198967?pr=4072):

> LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library [D:\a\sentry-dotnet\sentry-dotnet\integration-test\console-app\console-app.csproj]

I think that's good news as it means we can easily reproduce this now!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,7 @@
 ### Fixes
 
 - Prevent users from disabling AndroidEnableAssemblyCompression which leads to untrappable crash ([#4089](https://github.com/getsentry/sentry-dotnet/pull/4089))
+- MSVCRT build warning on Windows ([#4111](https://github.com/getsentry/sentry-dotnet/pull/4111))
 
 ## 5.5.1
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,7 +9,7 @@
 ### Fixes
 
 - Prevent users from disabling AndroidEnableAssemblyCompression which leads to untrappable crash ([#4089](https://github.com/getsentry/sentry-dotnet/pull/4089))
-- MSVCRT build warning on Windows ([#4111](https://github.com/getsentry/sentry-dotnet/pull/4111))
+- Fixed MSVCRT build warning on Windows ([#4111](https://github.com/getsentry/sentry-dotnet/pull/4111))
 
 ## 5.5.1
 

--- a/src/Sentry/buildTransitive/Sentry.targets
+++ b/src/Sentry/buildTransitive/Sentry.targets
@@ -16,7 +16,7 @@
   </Target>
 
   <ItemGroup Condition="$([System.OperatingSystem]::IsWindows())">
-    <!-- See: https://github.com/getsentry/sentry-dotnet/issues/3928 -->
+    <!-- See: https://github.com/getsentry/sentry-dotnet/pull/4111 -->
     <LinkerArg Include="/NODEFAULTLIB:MSVCRT" />
   </ItemGroup>
 

--- a/src/Sentry/buildTransitive/Sentry.targets
+++ b/src/Sentry/buildTransitive/Sentry.targets
@@ -15,6 +15,11 @@
     <Error Text="Android projects using Sentry cannot build using AndroidEnableAssemblyCompression = false due to a Microsoft issue.  Please follow https://github.com/dotnet/android/issues/9752" />
   </Target>
 
+  <ItemGroup>
+    <!-- See: https://github.com/getsentry/sentry-dotnet/issues/3928 -->
+    <LinkerArg Include="/NODEFAULTLIB:MSVCRT" />
+  </ItemGroup>
+
   <Target Name="WriteSentryAttributes"
           Condition="$(Language) == 'VB' or $(Language) == 'C#' or $(Language) == 'F#'"
           BeforeTargets="BeforeCompile;CoreCompile"

--- a/src/Sentry/buildTransitive/Sentry.targets
+++ b/src/Sentry/buildTransitive/Sentry.targets
@@ -15,7 +15,7 @@
     <Error Text="Android projects using Sentry cannot build using AndroidEnableAssemblyCompression = false due to a Microsoft issue.  Please follow https://github.com/dotnet/android/issues/9752" />
   </Target>
 
-  <ItemGroup>
+  <ItemGroup Condition="$([System.OperatingSystem]::IsWindows())">
     <!-- See: https://github.com/getsentry/sentry-dotnet/issues/3928 -->
     <LinkerArg Include="/NODEFAULTLIB:MSVCRT" />
   </ItemGroup>

